server/game/cards/12-KotI/ScoutingVessel.js

```diff
@@ -4,15 +4,15 @@ class ScoutingVessel extends DrawCard {
     setupCardAbilities(ability) {
         this.interrupt({
             when: {
-                onPillage: event => event.source.getType() === 'character' && event.source.controller === this.controller
+                onTopCardsDiscarded: event => event.source.getType() === 'character' && event.source.controller === this.controller && event.isPillage
             },
             cost: [
                 ability.costs.kneelSelf(),
                 ability.costs.sacrificeSelf()
             ],
             message: '{player} kneels and sacrifices {source} to discard 3 cards with pillage instead of 1',
             handler: context => {
-                context.event.numCards += 2;
+                context.event.amount += 2;
```

Comment:
Given how the Pillage event worked before vs the onTopCardsDiscarded event works, this can now increase the amount beyond the number of cards in the deck. It won't crash anything, e.g.:
```
> [1, 2].slice(0, 4)
[ 1, 2 ]
```

Would only be a problem if we ever have a reaction to the `onTopCardsDiscarded` event's amount, which should probably never happen - reactions should be to the cards themselves being discarded.